xbmc/filesystem/FileCache.cpp

```diff
@@ -277,13 +277,13 @@ void CFileCache::Process()
 
     while (m_writeRate)
     {
-      if (m_writePos - m_readPos < m_writeRate)
+      if (m_writePos - m_readPos < m_writeRate * g_advancedSettings.m_readBufferFactor)
```

Comment:
I wonder, where the readrate is set now, as you removed m_readBufferFactor from the SetReadRate above.